src/ci_workflow/ci_manifests.py

```diff
@@ -23,6 +26,29 @@ def __klass(filename: str) -> Union[Type[CiTestManifest], Type[CiInputManifest]]
         else:
             return CiInputManifest
 
+    @staticmethod
+    def __get_duplicate_component_names(count_component_names: Counter) -> list:
+        duplicate_component_names = []
+        for component_name, count in count_component_names.items():
+            if count > 1:
+                duplicate_component_names.append(component_name)
+        return duplicate_component_names
+
+    @staticmethod
+    def __check_duplicate_component_names(file: TextIOWrapper) -> None:
+        yaml_dict = yaml.safe_load(file)
+        component_names = []
+        for component in yaml_dict['components']:
+            component_names.append(component['name'])
+        count_component_names = Counter(component_names)
+
+        if set(count_component_names.values()) != set([1]):
+            duplicate_component_names = CiManifests.__get_duplicate_component_names(count_component_names)
+            duplicate_component_names_string = ', '.join(duplicate_component_names)
+            raise ValueError(f"Found {duplicate_component_names_string} as a duplicate component(s) in manifest {file.name}. ")
+        file.seek(0)
+
```

Comment:
I'm late to this PR, but this implementation seems unnecessarily complicated. 

I would load the items, then check with a 1-liner from https://stackoverflow.com/questions/9835762/how-do-i-find-the-duplicates-in-a-list-and-create-another-list-with-them.